test/e2e/specs/specs-playwright/wp-likes-spec.js

```diff
@@ -0,0 +1,64 @@
+/**
+ * External dependencies
+ */
+import assert from 'assert';
+import config from 'config';
+import {
+	BrowserHelper,
+	DataHelper,
+	LoginFlow,
+	NewPostFlow,
+	GutenbergEditorPage,
+	LikesComponent,
+} from '@automattic/calypso-e2e';
+
+/**
+ * Constants
+ */
+const mochaTimeOut = config.get( 'mochaTimeoutMS' );
+const host = DataHelper.getJetpackHost();
+const viewportName = BrowserHelper.getViewportName();
+const quote =
+	'The foolish man seeks happiness in the distance. The wise grows it under his feet.\n— James Oppenheim';
+
+describe( `[${ host }] Likes: (${ viewportName }) @parallel`, function () {
+	this.timeout( mochaTimeOut );
+
+	describe( 'New post', function () {
+		let gutenbergEditorPage;
+		let likesComponent;
+
+		it( 'Log in', async function () {
+			const loginFlow = new LoginFlow( this.page, 'gutenbergSimpleSiteUser' );
+			await loginFlow.login();
+		} );
+
+		it( 'Start new post', async function () {
+			const newPostFlow = new NewPostFlow( this.page );
+			await newPostFlow.newPostFromNavbar();
+		} );
+
+		it( 'Enter post title', async function () {
+			gutenbergEditorPage = await GutenbergEditorPage.Expect( this.page );
+			const title = DataHelper.randomPhrase();
+			await gutenbergEditorPage.enterTitle( title );
+			const enteredTitle = await gutenbergEditorPage.getTitle();
+			assert( title === enteredTitle );
```

Comment:
I think it would be good to stay consistent in the aspect of self-validating our helpers. In the case of i.e. `clickLike` helper we are making sure that the action brings the expected result (the "Like" button turns into the "Liked" button). Here, we shift that responsibility to the test itself. This could lead to a situation where this helper needs to be validated every time it's used in a test. What do you think?

Reply:
My answer here to the question of `self-validating our helpers` is going to be a 'sometimes, maybe'. 

[This SQA post](https://sqa.stackexchange.com/questions/11624/include-assertions-expectations-within-the-page-objects-yes-or-no) is a bit old, but it outlines that some assertions are better served in the page object while others are better in the tests.

[Another useful SQA post](https://sqa.stackexchange.com/questions/27751/my-understanding-are-correct-or-not-in-page-object-pattern/27765) references the `Arrange, Act, Assert` principle which advocates for assertions to be primarily in the tests.

So, in the end it seems to be a mixed bag. I felt there was a distinction between the title assertion and clickLike assertion, but maybe they are two sides of the same coin after all.

----
What if I move the assertion inside `GutenbergEditorPage.getTitle`, would it feel more consistent?

Reply:
I've restructured the API a bit.

`GutenbergEditorComponent.enterTitle()` now calls two distinct methods that act as a setter/getter. The setter sets the title which is then retrieved by the getter, and assertion takes place in the `enterTitle` method call before control is passed back to the caller.

This way, the setter is only concerned with filling in the text.